aopy/data/bmi3d.py

```diff
@@ -1716,6 +1969,284 @@ def tabulate_stim_data(preproc_dir, subjects, ids, dates, metadata=['stimulation
+            a smoothing function or an estimate of velocity from position
+        datatype (str, optional): type of kinematics to tabulate. Defaults to 'cursor'.  
+        kwargs (dict, optional): optional keyword arguments to pass to :func:`~aopy.preproc.get_kinematic_segment`  
+
+    Returns:
+        (ntrial,): list of tensors of (nt, nch) kinematics from each trial
+    '''
+
+    assert len(subjects) == len(te_ids) == len(dates) == len(start_times) == len(end_times)
+
+    segments = [_get_kinematic_segment(preproc_dir, s, t, d, ts, te, samplerate, preproc, datatype, **kwargs)[0] 
+                for s, t, d, ts, te in zip(subjects, te_ids, dates, start_times, end_times)]
+    trajectories = np.array(segments, dtype='object')
+    return trajectories
+
+def tabulate_task_data(preproc_dir, subjects, te_ids, dates, start_times, end_times, 
+                       datatype, samplerate=None, step=1, preproc=None, **kwargs):
+    '''
+    Grab task data from trials across arbitrary preprocessed files.
+
+    Args:
+        preproc_dir (str): base directory where the files live
+        subjects (list of str): Subject name for each recording
+        ids (list of int): Block number of Task entry object for each recording
+        dates (list of str): Date for each recording
+        datatype (str): column of task data to load. 
+        samplerate (float, optional): choose the samplerate of the data in Hz. Default None,
+            which uses the sampling rate of the experiment.
+        start_times (list of float): times in the recording at which the desired segments starts
+        end_times (list of float): times in the recording at which the desired segments ends
+        step (int, optional): integer step to slice the data. Default 1.
+        preproc (fn, optional): function mapping (position, fs) data to (kinematics, fs_new). For example,
+            a smoothing function or an estimate of velocity from position
+        kwargs: additional keyword arguments to pass to get_interp_task_data 
+    Returns:
+        tuple: tuple containing:
+            | **segments (ntrial,):** list of tensors of (nt, nch) task data from each trial
+            | **samplerate (float):** samplerate of the task data
+    '''
+
+    assert len(subjects) == len(te_ids) == len(dates) == len(start_times) == len(end_times)
+
+    segments = []
+    for s, t, d, ts, te in zip(subjects, te_ids, dates, start_times, end_times):
+        task_data, samplerate = get_task_data(
+            preproc_dir, s, t, d, datatype, samplerate=samplerate, step=step,
+            preproc=preproc, **kwargs) # cached for each unique recording
+        segments.append(get_data_segment(task_data, ts, te, samplerate))
+
+    segments = np.array(segments, dtype='object')
+    return segments, samplerate
+
+def tabulate_lfp_features(preproc_dir, subjects, te_ids, dates, start_times, end_times,
```

Comment:
Note that the same decoder is used for each te_id even though that may not have been what happened in practice. Also, it would be nice to be able to input multiple different decoders; maybe a list the same length as te_ids that has decoders for each te_id